src/python/pants/core/goals/test.py

```diff
@@ -406,6 +413,26 @@ def rules(cls) -> Iterable:
             yield from _unsupported_debug_adapter_rules(cls)
 
 
+@rule(polymorphic=True)
+async def partition_tests(req: TestRequest.PartitionRequest) -> Partitions:
+    raise NotImplementedError()
+
+
+@rule(polymorphic=True)
+async def to_debug_request(batch: TestRequest.Batch) -> TestDebugRequest:
+    raise NotImplementedError()
+
+
+@rule(polymorphic=True)
+async def to_debug_adapter_request(batch: TestRequest.Batch) -> TestDebugAdapterRequest:
```

Comment:
Naming: Consider whether this is better named as test_batch_to_debug_adapter_request?